integration-cli/docker_cli_swarm_test.go

```diff
@@ -946,10 +951,164 @@ func (s *DockerSwarmSuite) TestSwarmLeaveLocked(c *check.C) {
 	c.Assert(info.LocalNodeState, checker.Equals, swarm.LocalNodeStateActive)
 }
 
+func (s *DockerSwarmSuite) TestSwarmLockUnlockCluster(c *check.C) {
+	d1 := s.AddDaemon(c, true, true)
+	d2 := s.AddDaemon(c, true, true)
+	d3 := s.AddDaemon(c, true, true)
+
+	// they start off unlocked
+	c.Assert(d2.Restart(), checker.IsNil)
+	checkLockStatus(c, d2, false)
+
+	// stop this one so it does not get autolock info
+	c.Assert(d2.Stop(), checker.IsNil)
+
+	// enable autolock
+	outs, err := d1.Cmd("swarm", "update", "--autolock")
+	c.Assert(err, checker.IsNil, check.Commentf("out: %v", outs))
+
+	c.Assert(outs, checker.Contains, "docker swarm unlock")
+
+	var unlockKey string
+	for _, line := range strings.Split(outs, "\n") {
+		if strings.Contains(line, "SWMKEY") {
+			unlockKey = strings.TrimSpace(line)
+			break
+		}
+	}
+
+	c.Assert(unlockKey, checker.Not(checker.Equals), "")
+
+	outs, err = d1.Cmd("swarm", "unlock-key", "-q")
+	c.Assert(outs, checker.Equals, unlockKey+"\n")
+
+	// wait for everything to be locked
+	time.Sleep(1 * time.Second)
```

Comment:
I'm afraid the test will be fragile with these sleeps. Can we copy the strategy in `TestSwarmRotateUnlockKey` where we retry with a longer sleep if we didn't get the expected result?